app/connector/email/src/test/java/io/syndesis/connector/email/consumer/EMailReadRouteTest.java

```diff
@@ -0,0 +1,427 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.connector.email.consumer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import org.apache.camel.Endpoint;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.mail.MailEndpoint;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.support.DependencyInjectionTestExecutionListener;
+import org.springframework.test.context.support.DirtiesContextTestExecutionListener;
+import io.syndesis.common.model.DataShape;
+import io.syndesis.common.model.DataShapeKinds;
+import io.syndesis.common.model.action.ConnectorAction;
+import io.syndesis.common.model.action.ConnectorDescriptor;
+import io.syndesis.common.model.connection.Connector;
+import io.syndesis.common.model.integration.Integration;
+import io.syndesis.common.model.integration.Step;
+import io.syndesis.common.util.SuppressFBWarnings;
+import io.syndesis.connector.email.AbstractEMailRouteTest;
+import io.syndesis.connector.email.component.EMailComponentFactory;
+import io.syndesis.connector.email.customizer.EMailReceiveCustomizer;
+import io.syndesis.connector.email.model.EMailMessageModel;
+import io.syndesis.connector.email.server.EMailTestServer;
+import io.syndesis.connector.support.util.PropertyBuilder;
+
+@DirtiesContext
+@RunWith(SpringRunner.class)
+@SpringBootTest(
+    classes = {
+        EMailReadRouteTest.TestConfiguration.class
+    },
+    properties = {
+        "spring.main.banner-mode = off",
+        "logging.level.io.syndesis.integration.runtime = DEBUG"
+    }
+)
+@TestExecutionListeners(
+    listeners = {
+        DependencyInjectionTestExecutionListener.class,
+        DirtiesContextTestExecutionListener.class
+    }
+)
+@SuppressFBWarnings
+public class EMailReadRouteTest extends AbstractEMailRouteTest {
+
+    private EMailTestServer server;
+
+    public EMailReadRouteTest() throws Exception {
```

Comment:
needed, just calls `super()` ?

Reply:
Was due to a 'throws Exception" in super constructor. Now removed.